Eastern-Research-Group/mywaterway
# Feature/finish moving state lookups to s3

## Related Issues:
* https://app.breeze.pm/projects/100762/cards/3411201

## Main Changes:
* Converted all references to documentOrder.js file into using the new custom hooks for fetching the documentOrder.json file from the proxy/s3.
* Converted all references to introText.js file into using the new custom hooks for fetching the introText.json file from the proxy/s3. 
* Converted all references to reportStatusMapping.js file into using the new custom hooks for fetching the reportStatusMapping.json file from the proxy/s3.
* Converted all references to stateNationalUses.js file into using the new custom hooks for fetching the stateNationalUses.json file from the proxy/s3. 

## Steps To Test:
1. Navigate to http://localhost:3000/state/KS/water-quality-overview
2. Open the chrome dev tools.
3. On the network tab type "proxy" into the filter.
4. Verify the "introText", "reportStatusMapping", and "stateNationalUses" files were only fetched one time using the proxy.
5. Verify the state intro text, state metrics, and link to the state's site (bottom of page) are displayed.
6. Verify the water type and use dropdowns have data.
7. Block the "introText" fetch.
8. Reload the page.
9. Verify in error message is displayed where the intro text and metrics would be displayed. 
10. Unblock the "introText" fetch and block the "stateNationalUses" fetch.
11. Reload the page.
12. Verify the error message is displayed.
13. Unblock the "introText" fetch. 
14. Reload the page.
15. Expand the documents section.
16. Verify the "documentOrder" file is only fetched one time using the proxy. 
17. Verify the ordering is based on the numeric ordering defined in the "documentOrder" lookup.
18. Block the "documentOrder" fetch.
19. Reload the page.
20. Verify the "There was an issue sorting..." info message is displayed.
21. Verify the documents are ordered alphabetically by the document types column.
22. Unblock the "documentOrder" fetch. 
23. Navigate to http://localhost:3000/state/IN/advanced-search
24. Select one waterbody from the "Waterbody Names" select and click search.
25. Verify the map footer says "Submitted to EPA for Review"
26. Block the "reportStatusMapping" fetch
27. Reload the page.
28. Select one waterbody from the "Waterbody Names" select and click search.
29. Verify the map footer has an info message and says "Organization Final Action - Submittal".
